src/rdb.c

```diff
@@ -1020,6 +1033,13 @@ ssize_t rdbSaveObject(rio *rdb, robj *o, robj *key, int dbid) {
                     return -1;
                 }
                 nwritten += n;
+                
+                /* Save the group's logical reads counter. */
+                if ((n = rdbSaveLen(rdb,cg->entries_read)) == -1) {
+                    raxStop(&ri);
+                    return -1;
+                }
```

Comment:
entries_read can be `SCG_INVALID_ENTRIES_READ`, which is -1.
and here `rdbSaveLen` take a `uint64_t` input, so what is actually written here is another value, such as 18446744073709551615

and then when we load this `entries_read` back, we will convert it from `uint64_t` to `long long`, it just happened to be -1 again, so everything was fine. Do we need to make changes here?

also, i found that the entries_read of the maste and the replica can be different. This should be expected, right? since it is not an real "data"

Reply:
@guybe7 recently pointed out to me that RM_SaveSigned has a similar problem (and that RDB_MODULE_OPCODE_SINT is unused).
the implication is that unlike rdbEncodeInteger (used for values of keys), these are less efficient in their file size consumption.

fixing these would mean a new rdb format, and for new versions to be able to properly handle the old one.
since unstable already has an rdb version bump, we can try tackling these two issues if we want.
not a real priority IMHO.

Reply:
> since unstable already has an rdb version bump

The most recent one was in #11099, released in 7.2. yes, it is not a real priority, i have some spare time these day, just ping me if you want me to fix it

Reply:
the main benefit for that would be disk space reduction for modules that heavily use RM_SaveSigned.
if you have time you can try to see if this gets complicated (including tests for backwards compatibility).